package/MDAnalysis/coordinates/XDR.py

```diff
@@ -141,9 +157,18 @@ def _reopen(self):
 
     def _read_frame(self, i):
         """read frame i"""
-        self._xdr.seek(i)
         self._frame = i - 1
-        return self._read_next_timestep()
+        try:
+            self._xdr.seek(i)
+            timestep = self._read_next_timestep()
+        except RuntimeError:
+            warnings.warn('seek failed, recalculating offsets and retry')
+            offsets = self._xdr.calc_offsets()
+            self._xdr.set_offsets(offsets)
+            self._read_offsets(store=True)
+            self._xdr.seek(i)
+            timestep = self._read_next_timestep()
+        return timestep
```

Comment:
Adding the check in `_read_frame` is fragile. There is at least one other point in the code that calls `_read_next_timestep()` directly (for instance, the `next()/__next__()` methods defined in `base.py`). Better check inside `_read_next_timestep()`.